keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/CombinedRepository.kt

```diff
@@ -91,9 +92,29 @@ class CombinedRepository(
     deliveryConfig.resources.forEach { resource ->
       upsert(resource)
     }
+
     deliveryConfig.artifacts.forEach { artifact ->
       register(artifact)
     }
+
+    deliveryConfig.resources.forEach { resource ->
+      upsert(resource)
+      if (resource.spec is ComputeResourceSpec) {
+        val computeResource = resource as Resource<ComputeResourceSpec>
+        val artifactReference = computeResource.spec.completeArtifactReferenceOrNull()
+        if (artifactReference != null) {
+          val artifact = deliveryConfig.matchingArtifactByReference(
+            artifactReference.artifactReference,
+            artifactReference.artifactType
+          )
+          if (artifact != null) {
+            log.debug("Associating resource ${computeResource.id} with artifact $artifact")
+            resourceRepository.associate(computeResource, artifact)
+          }
+        }
+      }
+    }
```

Comment:
This is where we use the new artifact interfaces (here represented by `ComputeResourceSpec`) to look up an artifact to associate with the resource in the database. Only reference-based artifact providers are supported, as discussed/agreed.